gwpy/cli/cliproduct.py

```diff
@@ -469,21 +469,29 @@ def _filter_timeseries(data, highpass=None, lowpass=None, notch=None):
         if all(x is None for x in (highpass, lowpass, notch)):
             return data
 
-        # build ZPK
-        zpks = []
+        # apply sos filters one at a time to avoid problems combining them
+
+        sos = None
         if highpass is not None and lowpass is not None:
-            zpks.append(filter_design.bandpass(highpass, lowpass,
-                                               data.sample_rate))
+            sos = filter_design.bandpass(highpass, lowpass, data.sample_rate,
+                                         ftype='butter', output='sos')
         elif highpass is not None:
-            zpks.append(filter_design.highpass(highpass, data.sample_rate))
+            sos = filter_design.highpass(highpass, data.sample_rate,
+                                         ftype='butter', output='sos')
         elif lowpass is not None:
-            zpks.append(filter_design.lowpass(lowpass, data.sample_rate))
+            sos = filter_design.lowpass(lowpass, data.sample_rate,
+                                        ftype='butter', output='sos')
+
+        if sos is not None:
+            ret = data.filter(sos)
+        elif notch:
+            ret = data
+
         for freq in notch or []:
-            zpks.append(filter_design.notch(freq, data.sample_rate))
-        zpk = filter_design.concatenate_zpks(*zpks)
+            sos = filter_design.notch(freq, data.sample_rate)
```

Comment:
ATOW the `filter_design.notch()` actually _always_ returns a ZPK. But I am preparing a patch to enable opt-in to `sos`, please hold.